### PR TITLE
I can bookmark a racipe

### PR DESCRIPTION
**Pull request by Za-hell**
> I can create
> I can see 
> Missing I can destroy 
Screenshot from 2023-03-20 20-11-23

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -5,4 +5,6 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   has_one :fridge, dependent: :destroy
   validates :phone_number, format: { with: /\A(?:\+?\d{1,3}\s*-?)?\(?(?:\d{3})?\)?[- ]?\d{3}[- ]?\d{4}\z/ }
+  has_many :favorite_recipes
+  has_many :recipes, through: :favorite_recipes
```
Comment (akramerin) [inline app/models/user.rb:9]: i am not sure about that - should it not be: has_many :recipes (as in the ones that he created) additionally to the current line nine?

also: still the old (better :D ) spelling

File: app/views/favourite_recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<div>
+  <%= current_user.favourite_recipes.each do |p| %>
+  <div class ="row hoverable posts-background">
+    <div class ="col s10">
+    <%= link_to "#{p.name}", favorite_recipes(p)%>
```
Comment (akramerin) [inline app/views/favourite_recipes/index.html.erb:5]: spelling

File: app/views/recipes/index.html.erb
Comment (syboulette) [inline app/views/recipes/index.html.erb:1]: @Za-hell can we use the term favourite please? I feel like book mark is not super nice, also we always talked about favorites. Please stick to what we agreed on at the begining ;)

File: app/views/recipes/index.html.erb [hunk 3]
```diff
@@ -14,3 +14,13 @@
     </div>
   <% end %>
 </div>
+
+
+<%# BOOKMARKS %>
+<%= simple_form_for @favourite_recipe do |f| %>
+  <% if current_user %>
+    <%= f.submit "Bookmark", class: "btn btn-danger pic-large", id: "buttonz" %>
```
Comment (syboulette) [inline app/views/recipes/index.html.erb:22]: Can we have a little heart that fills up when marked as favourite like we have on the figma? 🙏

Comment (syboulette) [reply app/views/recipes/index.html.erb:22]: a big button on the card will not work

File: db/schema.rb [hunk 4]
```diff
@@ -69,6 +69,24 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "favorites", force: :cascade do |t|
```
Comment (syboulette) [inline db/schema.rb:72]: @Za-hell can we keep the "favorite" spelling across the code as it is the one you decided on for the schema? thanks!

File: app/controllers/favourite_recipes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class FavouriteRecipesController < ApplicationController
+  before_action :authenticate_user!, only: :create
```
[the remark this answers is not in the document]
Comment (syboulette) [reply app/controllers/favourite_recipes_controller.rb:2]: @Za-hell keep it consistant with the schema, please use "Favorite"

File: app/controllers/favourite_recipes_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  def create
+    @favourite_recipe = FavouriteRecipe.new(favourite_recipe_params)
```
Comment (syboulette) [inline app/controllers/favourite_recipes_controller.rb:4]: @Za-hell Favorite

File: app/controllers/recipes_controller.rb [hunk 7]
```diff
@@ -4,6 +4,7 @@ class RecipesController < ApplicationController
 
   def index
     @recipes = policy_scope(Recipe).all
+    @favourite_recipe = FavouriteRecipe.new
```
Comment (syboulette) [inline app/controllers/recipes_controller.rb:7]: @Za-hell Favorite

File: app/views/favourite_recipes/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<div>
+  <%= current_user.favourite_recipes.each do |p| %>
```
Comment (syboulette) [inline app/views/favourite_recipes/index.html.erb:2]: @Za-hell Favorite

File: app/views/recipes/index.html.erb [hunk 9]
```diff
@@ -14,3 +14,13 @@
     </div>
   <% end %>
 </div>
+
+
+<%# BOOKMARKS %>
+<%= simple_form_for @favourite_recipe do |f| %>
```
Comment (syboulette) [inline app/views/recipes/index.html.erb:20]: @Za-hell Favorite

File: config/routes.rb [hunk 10]
```diff
@@ -11,18 +11,18 @@
     resources :recipe_ingredients, only: [:edit, :update]
   end
   resources :recipe_ingredients, only: [:new, :create, :destroy]
-  resources :favorite_recipes, only: [:new, :create, :edit, :index]
+  resources :favourite_recipes, only: [:new, :create, :edit, :index]
```
Comment (syboulette) [inline config/routes.rb:14]: @Za-hell Favorite

File: test/controllers/favorite_recipes_controller_test.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class FavoriteRecipesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
```
Comment (syboulette) [inline test/controllers/favorite_recipes_controller_test.rb:4]: @Za-hell You created routes for new, create, index,edit and update. I think we also need a destroy as people might not want the recipe in their favourites anymore.
Also need the actions in here. Thanks!
S

**Review by syboulette — CHANGES_REQUESTED** [review]

**Pull request closed without merging**